packages/storage/src/service.ts

```diff
@@ -262,6 +266,9 @@ export class FirebaseStorageImpl implements FirebaseStorage {
   }
 
   async _getAppCheckToken(): Promise<string | null> {
+    if (_isFirebaseServerApp(this.app) && this.app.settings.appCheckToken) {
+      return this.app.settings.appCheckToken;
+    }
```

Comment:
Similiar to the other SDKs (above) check to see if the provided instance of `FirebaseApp` is one of `FirebaseServerApp`, and if it contains an App Check token, then return it instead of invoking the App Check SDK.